### ISSUE TITLE
[synthetic-shadow] Duplicate event listeners on template and host are not discarded

### ISSUE THREAD

**Issue by ravijayaramappa** [salesforce/lwc#2253]
### Description

As per DOM spec, if an event listener(with event type and options) is added more than once on the same element, the subsequent addEventListener() are discarded. https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#multiple_identical_event_listeners

However, duplicate listeners added to a template and host element are not being discarded.

#### Steps to Reproduce

```html
<template>
    <div >
        <span></span>
    </div>
    <button onclick={dispatchEvent}>Dispatch Event</button>
</template>
```
```js
import { LightningElement, track } from 'lwc';

function listener(event) {
    console.log("Event logged", event.currentTarget.tagName);
}
export default class App extends LightningElement {
    renderedCallback() {
        this.template.addEventListener('foo', listener);
        this.template.addEventListener('foo', listener);
        
        const div = this.template.querySelector('div');
        div.addEventListener('foo', listener);
        div.addEventListener('foo', listener);

        const span = this.template.querySelector('span');
        span.addEventListener('foo', listener);
        span.addEventListener('foo', listener);
    }

    dispatchEvent() {
        this.template.querySelector('span').dispatchEvent(new CustomEvent('foo', {bubbles: true}));
    }
}

```
https://playground.lwcjs.org/projects/NI2OWuVnO/3/edit

To reproduce the bug on host element, make these changes

```
 this.addEventListener('foo', listener);
this.template.querySelector('span').dispatchEvent(new CustomEvent('foo', {bubbles: true, composed: true}));
```

#### Expected Results
Expect the listener on this.template to be invoked only once

#### Actual Results
Listeners on standard elements invoked once as expected. Listeners on shadow root and host element invoked multiple times.

### Browsers Affected
Chrime, Firefox

### Version
- LWC: 1.17.4

**Comment by ravijayaramappa** [comment]
@ekashida @Sumitjg Is this issue potentially solved by https://github.com/salesforce/lwc/pull/2226?

**Comment by ekashida** [comment]
@ravijayaramappa no, this issue will not be fixed by #2226
